pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,762 @@
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithUID(podUID).
+						WithController(true)).
+					WithStatus(resourcev1alpha1ac.ResourceClaimStatus().
+						WithAllocation(resourcev1alpha1ac.AllocationResult().
+							WithAvailableOnNodes(corev1ac.NodeSelector().
+								WithNodeSelectorTerms(corev1ac.NodeSelectorTerm().
+									WithMatchExpressions(corev1ac.NodeSelectorRequirement().
+										WithKey("nodename").
+										WithOperator(v1.NodeSelectorOpIn).
+										WithValues("worker")))))))
+	otherClaim = toResourceClaim(resourcev1alpha1ac.ResourceClaim("not-my-claim", namespace))
+
+	schedulingSelectedPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithAPIVersion("v1").
+						WithController(true).
+						WithKind("Pod").
+						WithName(podName).
+						WithUID(podUID)).
+					WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+						WithSelectedNode(workerNode.Name).
+						WithPotentialNodes(workerNode.Name)))
+	schedulingPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion("v1").
+					WithController(true).
+					WithKind("Pod").
+					WithName(podName).
+					WithUID(podUID)).
+				WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+					WithPotentialNodes(workerNode.Name)))
+	schedulingInfo = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+			WithOwnerReferences(metav1ac.OwnerReference().
+				WithAPIVersion("v1").
+				WithController(true).
+				WithKind("Pod").
+				WithName(podName).
+				WithUID(podUID)).
+			WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+				WithPotentialNodes(workerNode.Name)).
+			WithStatus(resourcev1alpha1ac.PodSchedulingStatus().
+				WithResourceClaims(resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName), resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName2))))
+)
+
+// result defines the expected outcome of some operation. It covers return
+// values, the plugin state and the state of the world (= objects).
+type result struct {
+	returnValue interface{}
+	status      *framework.Status
+
+	// changes contains corev1ac apply configuration instances that contain
+	// the expected changes to existing objects. This way, new fields and
+	// can be described concisely. Removal of fields is not needed.
+	changes []any
+
+	// added contains objects created by the operation.
+	added []metav1.Object
+
+	// removed contains objects deleted by the operation.
+	removed []metav1.Object
+}
+
+type operation int
+
+const (
+	prefilterOp operation = iota
+	filterOp
+	prescoreOp
+	reserveOp
+	unreserveOp
+	postbindOp
+)
+
+type perNodeResult map[string]result
+
+type want map[operation]any
+
+func TestPlugin(t *testing.T) {
+	var nilPreFilterResult *framework.PreFilterResult
+
+	testcases := map[string]struct {
```

Comment:
the list of testcases looks surprisingly small. What is the coverage you get?

Reply:
70%. The volumebinding plugin is at 82%.

Both could be higher, but for alpha I thought that this would be good enough.

Reply:
I'm already hearing a lot of "good enough for alpha" that I'm growing more nervous about this PR.

Reply:
How high do you want test coverage to be?

Reply:
> I'm already hearing a lot of "good enough for alpha" that I'm growing more nervous about this PR.

Should all those places have "re-review for beta" TODO comments?

Reply:
I think the KEP checklist for beta is covering this. It explicitly asks for things like test coverage, list of E2E tests, etc., so we will come back here with or without TODO comments.